src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java

```diff
@@ -122,6 +123,18 @@ public void testTypecast() throws Exception {
                 getPath("InputRegressionJavaTypecast.java"));
     }
 
+    @Test
+    public void testCharLiteralSurrogatePair() throws Exception {
+        if (Charset.isSupported("windows-1252")) {
```

Comment:
Please look at `AbstractPathTestSupport.readFile` and ascertain why the file isn't being read in UTF8.

Reply:
So, should we default to UTF-8, then?

Reply:
We are already trying to read the files as UTF8 at https://github.com/checkstyle/checkstyle/blob/ae9edbd4a8645c8cc1a99ad6890dd0006700d1cb/src/test/java/com/puppycrawl/tools/checkstyle/AbstractPathTestSupport.java#L80 . Unless you need a different file format, I think we should force UTF8.

Reply:
https://github.com/checkstyle/checkstyle/pull/7886#issuecomment-601468198